### PR TITLE
fix: support undefined process

### PR DESCRIPTION
**Pull request by cazala**
There are builds where `process` is not defined and is making the library to throw on runtime with the following error: 

```
TypeError: Cannot read properties of undefined (reading 'env')

      120 |  */
      121 | export function getEnv(
    > 122 |   systemEnvVariables: EnvironmentVariables = process.env
```

This PR adds a test that reproduces the issue, and then applies a fix for it.

### CODE REVIEW

File: src/env.spec.ts [hunk 1]
```diff
@@ -249,10 +251,17 @@ describe('when getting env', () => {
...
+        mockProcessEnv({ DCL_DEFAULT_ENV: 'dev' })
         expect(getEnv()).toBe(Env.STAGING)
       })
     })
+
+    describe('and process is not defined', () => {
+      it('should return Env.PRODUCTION', () => {
+        process = undefined as any
+        expect(getEnv()).toBe(Env.PRODUCTION)
+      })
```
Comment (cazala) [inline src/env.spec.ts:294]: This tests reproduces the issue

File: src/env.ts [hunk 2]
```diff
@@ -133,5 +131,12 @@ export function getEnv(
     }
   }
 
-  return getDefaultEnv(systemEnvVariables)
+  const envVars =
+    typeof systemEnvVariables === 'undefined' &&
+    typeof process === 'object' &&
+    typeof process.env === 'object'
+      ? process.env
+      : systemEnvVariables
```
Comment (cazala) [inline src/env.ts:139]: This should keep the same behaviour for projects relying on the `process.env` and passing no param to the `getEnv` helper.

If no param is sent to the helper, and `process.env` exists, it uses that as the `envVars`, otherwise if something custom is passed as param, it will use the custom param.

All the existing tests are working without having to make any changes on the tests themselves.

File: src/env.spec.ts [hunk 3]
```diff
@@ -168,6 +168,29 @@ describe('when getting default env', () => {
...
+      it('should throw', () => {
+        expect(() =>
+          getDefaultEnv({
+            DCL_DEFAULT_ENV: 'dev',
+            VITE_DCL_DEFAULT_ENV: 'prod',
+          })
+        ).toThrow()
+      })
+    })
+  })
```
Comment (cazala) [inline src/env.spec.ts:192]: This test is not related to this PR, it was just missing and causing the coverage to be less than 100% so I added it.

File: src/env.spec.ts [hunk 4]
```diff
@@ -269,7 +309,7 @@ describe('when getting env', () => {
     })
   })
 
-  describe('and host is "market.decentraland.org"', () => {
+  describe('and host is "builder.decentraland.org"', () => {
```
Comment (cazala) [inline src/env.spec.ts:312]: This was just a typo in the test name

**Review by gonpombo8 — APPROVED** [review]
